From 0d9b6a3ff1b31ce98d8e07ee1cae95435328fb29 Mon Sep 17 00:00:00 2001
From: Tim Snyder <timothy.snyder@sifive.com>
Date: Wed, 29 Jun 2022 14:47:04 +0000
Subject: [PATCH 1/3] Add explicit -lrt

newer toolchains have changed to require -lrt to be explicitly mentioned where
it used to be impliictly linked via one of the stdlibs
---
 Makefile             |  3 ++-
 tools/wake/wake.wake |  2 +-
 vendor/rt.wake       | 19 +++++++++++++++++++
 3 files changed, 22 insertions(+), 2 deletions(-)
 create mode 100644 vendor/rt.wake

diff --git a/Makefile b/Makefile
index cbbad6bb..be299c62 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,8 @@ FUSE_LDFLAGS := $(shell pkg-config --silence-errors --libs fuse    || echo -lfus
 CORE_LDFLAGS :=	$(shell pkg-config --silence-errors --libs sqlite3 || echo -lsqlite3)	\
 		$(shell pkg-config --silence-errors --libs gmp-6   || echo -lgmp)	\
 		$(shell pkg-config --silence-errors --libs re2     || echo -lre2)	\
-		$(shell pkg-config --silence-errors --libs ncurses tinfo || pkg-config --silence-errors --libs ncurses || echo -lncurses)
+		$(shell pkg-config --silence-errors --libs ncurses tinfo || pkg-config --silence-errors --libs ncurses || echo -lncurses) \
+		-lrt
 
 COMMON_DIRS := src/compat src/util src/json
 COMMON_C    := $(foreach dir,$(COMMON_DIRS),$(wildcard $(dir)/*.c)) \
diff --git a/tools/wake/wake.wake b/tools/wake/wake.wake
index 2edeb41e..66a0db71 100644
--- a/tools/wake/wake.wake
+++ b/tools/wake/wake.wake
@@ -17,4 +17,4 @@ package build_wake
 from wake import _
 
 target buildWake variant =
-    tool here Nil variant "bin/wake" (runtime, gopt, Nil)
+    tool here Nil variant "bin/wake" (runtime, gopt, rt, Nil)
diff --git a/vendor/rt.wake b/vendor/rt.wake
new file mode 100644
index 00000000..a1e96600
--- /dev/null
+++ b/vendor/rt.wake
@@ -0,0 +1,19 @@
+# Copyright 2019 SiFive, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You should have received a copy of LICENSE.Apache2 along with
+# this software. If not, you may obtain a copy at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package build_wake
+from wake import _
+
+def rt _ = Pass (pkg "rt")
-- 
2.35.3

